apps-rendering/src/components/WithAgeWarning/index.tsx

```diff
@@ -0,0 +1,185 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css } from '@emotion/react';
+import type { Tag } from '@guardian/content-api-models/v1/tag';
+import { ArticleDesign, ArticleDisplay } from '@guardian/libs';
+import type { ArticleFormat } from '@guardian/libs';
+import { from, remSpace } from '@guardian/source-foundations';
+import { AgeWarning } from '@guardian/source-react-components-development-kitchen';
+import type { Option } from '@guardian/types';
+import { OptionKind } from '@guardian/types';
+import { grid } from 'grid/grid';
+import type { Item } from 'item';
+import { isComment, isNews } from 'item';
+import { articleWidthStyles, wideContentWidth } from 'styles';
+
+interface WithAgeWarningProps {
+	item: Item;
+	publishDate: Option<Date>;
+	format: ArticleFormat;
+}
+
+const getAgeWarning = (
+	tags: Tag[],
+	publicationDate: Date,
+): string | undefined => {
+	const isNewsArticle = isNews(tags);
+	const isOpinion = isComment(tags);
+	let message;
+
+	// Only show an age warning for news or opinion pieces
+	if (isNewsArticle || isOpinion) {
+		const warnLimitDays = 30;
+		const currentDate = new Date();
+		const dateThreshold = new Date();
+
+		dateThreshold.setDate(currentDate.getDate() - warnLimitDays);
+
+		// if the publication date is before the date threshold generate message
+		if (publicationDate < dateThreshold) {
+			// Unary + coerces dates to numbers for TypeScript
+			const diffMilliseconds = +currentDate - +publicationDate;
+			const diffSeconds = Math.floor(diffMilliseconds / 1000);
+			const diffMinutes = diffSeconds / 60;
+			const diffHours = diffMinutes / 60;
+			const diffDays = diffHours / 24;
+			const diffMonths = diffDays / 31;
+			const diffYears = diffDays / 365;
+
+			if (diffYears >= 2) {
+				message = `${Math.floor(diffYears)} years old`;
+			} else if (diffYears > 1) {
+				message = '1 year old';
+			} else if (diffMonths >= 2) {
+				message = `${Math.floor(diffMonths)} months old`;
+			} else if (diffMonths > 1) {
+				message = '1 month old';
+			}
+		}
+	}
+
+	return message;
+};
+
+export const defaultWidthStyles: SerializedStyles = css`
+	${from.wide} {
+		margin: 0 auto;
+	}
+
+	${from.phablet} {
+		width: ${wideContentWidth}px;
+	}
+`;
+
+export const warningStyles = (
+	format: ArticleFormat,
+	isSeries: boolean,
+): SerializedStyles => {
+	switch (format.design) {
+		case ArticleDesign.Gallery:
+			return css`
+				${galleryStyle(isSeries)}
+			`;
```

Comment:
I don't think you need to make a second call to `css` here, the `galleryStyle` function already returns `SerializedStyles`.

```suggestion
			return galleryStyle(isSeries);
```